worlds/kh1/__init__.py

```diff
@@ -0,0 +1,314 @@
+            elif name == "High Jump":
+                if self.options.extra_shared_abilities:
+                    item_pool += [self.create_item(name) for _ in range(0, 3)]
+                else:
+                    item_pool += [self.create_item(name) for _ in range(0, quantity)]
+            elif name == "Progressive Glide":
+                if self.options.extra_shared_abilities:
+                    item_pool += [self.create_item(name) for _ in range(0, 4)]
+                else:
+                    item_pool += [self.create_item(name) for _ in range(0, quantity)]
+            elif name == "End of the World":
+                if self.options.end_of_the_world_unlock.current_key == "item":
+                    item_pool += [self.create_item(name) for _ in range(0, quantity)]
+            elif name == "EXP Zero":
+                if self.options.exp_zero_in_pool:
+                    item_pool += [self.create_item(name) for _ in range(0, quantity)]
+            elif name not in prefilled_items:
+                item_pool += [self.create_item(name) for _ in range(0, quantity)]
+        
+        for i in range(self.determine_reports_in_pool()):
+            item_pool += [self.create_item("Ansem's Report " + str(i+1))]
+        
+        while len(item_pool) > total_locations:
+            item_pool.pop(0)
+        
+        while len(item_pool) < total_locations and len(level_up_item_pool) > 0:
+            item_pool += [self.create_item(level_up_item_pool.pop())]
+        
+        # Fill any empty locations with filler items.
+        item_names = []
+        attempts = 0  # If we ever try to add items 200 times, and all the items are used up, lets clear the item_names array, we probably don't have enough items
+        while len(item_pool) < total_locations:
+            item_name = self.get_filler_item_name()
+            if item_name not in item_names:
+                item_names.append(item_name)
+                item_pool.append(self.create_item(item_name))
+                attempts = 0
+            elif attempts >= 200:
+                item_names = []
+                attempts = 0
+            else:
+                attempts = attempts + 1
+
+        self.multiworld.itempool += item_pool
+
+    def pre_fill(self) -> None:
+        goal_dict = {
+            "sephiroth":       "Olympus Coliseum Defeat Sephiroth Ansem's Report 12",
+            "unknown":         "Hollow Bastion Defeat Unknown Ansem's Report 13",
+            "postcards":       "Traverse Town Mail Postcard 10 Event",
+            "final_ansem":     "Final Ansem",
+            "puppies":         "Traverse Town Piano Room Return 99 Puppies Reward 2",
+            "final_rest":      "End of the World Final Rest Chest"
+        }
+        self.multiworld.get_location(goal_dict[self.options.goal.current_key], self.player).place_locked_item(self.create_item("Victory"))
+        if self.options.junk_in_missable_locations:
```

Comment:
I know I said this in `options.py` already, but missable locations really should not exist. These locations should not be created at all if `accessibility` is set to `locations` or `items`